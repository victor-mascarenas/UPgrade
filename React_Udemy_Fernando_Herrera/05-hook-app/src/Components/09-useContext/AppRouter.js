import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { AboutPage } from './AboutPage';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <div>
                <NavBar/>
                <div className='container'>
                    <Routes>
                        <Route exact path="/" element={<HomePage/>}/>
                        <Route exact path="about" element={<AboutPage/>}/>
                        <Route exact path="login" element={<LoginPage/>}/>
                    </Routes>
                </div>
            </div>
        </BrowserRouter>
    );
};
