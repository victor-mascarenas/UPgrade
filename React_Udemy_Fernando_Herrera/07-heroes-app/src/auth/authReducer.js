/* const state = {
    name: 'Victor',
    logged: true
}; */

import { types } from "../types/types";

export const authReducer = (state = {}, action) => {
    switch(action.type) {
        case types.login:
            state = {
                name: action.payload.name,
                logged: true
            };
            break;
        case types.logout:
            state = {
                logged: false
            }
            break;
        default:
            break;
    }
    return state;
};