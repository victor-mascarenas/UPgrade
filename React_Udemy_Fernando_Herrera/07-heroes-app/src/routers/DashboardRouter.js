import React from 'react';
import { Route, Routes } from 'react-router-dom';
import { Navbar } from '../components/ui/Navbar';
import { MarvelPage } from '../components/marvel/MarvelPage';
import { SearchPage } from '../components/search/SearchPage';
import { DcPage } from '../components/dc/DcPage';
import { HeroPage } from '../components/hero/HeroPage';

export const DashboardRouter = () => {
    return (
        <>
            <Navbar/>
            <div className='container'>
                <Routes>
                    <Route path='marvel' element={<MarvelPage/>}/>
                    <Route path='dc' element={<DcPage/>}/>
                    <Route path='search' element={<SearchPage/>}/>
                    <Route path='hero/:heroId' element={<HeroPage/>}/>
                    <Route path='/' element={<MarvelPage/>}/>
                </Routes>
            </div>
        </>
    );
};
