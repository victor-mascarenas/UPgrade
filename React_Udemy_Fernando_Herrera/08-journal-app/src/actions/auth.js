import { types } from "../reducers/types/types";
import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { endLoading, startLoading } from "./ui";
import Swal from 'sweetalert2';
import { notesLogout } from "./notes";

export const startLogin = (email, password) => {
    return (dispatch) => {//dispatch comes from Thunk
        dispatch(startLoading());
        return firebase.auth().signInWithEmailAndPassword(email, password)
            .then(({user}) => {
                dispatch(login(user.uid, user.displayName));
            })
            .catch(error => {
                Swal.fire('Error', error.message, 'error');
            })
            .finally(() => {
                dispatch(endLoading());
            });
    };
};

export const startCredentialsRegistration = (email, password, name) => {
    return (dispatch) => {
        firebase.auth().createUserWithEmailAndPassword(email, password)
            .then(async ({user}) => {
                await user.updateProfile({
                    displayName: name
                });
                dispatch(login(user.uid, user.displayName));
            })
            .catch(error => {
                Swal.fire('Error', error.message, 'error');
            });
    };
};

export const startGoogleLogin = () => {
    return (dispatch) => {
        firebase.auth().signInWithPopup(googleAuthProvider)
            .then(({user}) => {
                dispatch(login(user.uid, user.displayName));
            })
            .catch((error) => {
                Swal.fire('Error', error.message, 'error');
            });
    };
};

export const login = (uid, displayName) => (
    {
        type: types.LOGIN,
        payload: {
            uid,
            displayName
        }
    }
);

export const startLogout = () => {
    return async (dispatch) => {
        await firebase.auth().signOut();
        dispatch(logout());
        dispatch(notesLogout());
    };
};

export const logout = () => {
    return {
        type: types.LOGOUT
    };
}