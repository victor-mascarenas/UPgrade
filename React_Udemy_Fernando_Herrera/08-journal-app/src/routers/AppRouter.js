import React from 'react'
import { BrowserRouter, Route, Routes, Navigate } from 'react-router-dom'
import { JournalScreen } from '../components/journal/JournalScreen'
import { AuthRouter } from './AuthRouter'

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route path='/auth/*' element={<AuthRouter/>}/>
                <Route exact path='/' element={<JournalScreen/>}/>
                <Route path='*' element={<Navigate to='/auth/login'/>}/>
            </Routes>
        </BrowserRouter>
    );
}
