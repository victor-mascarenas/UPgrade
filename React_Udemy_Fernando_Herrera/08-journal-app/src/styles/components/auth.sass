.auth__main
    align-items: center
    background-color: $primary
    display: flex
    justify-content: center
    height: 100vh
    margin: 0px
    width: 100vw
.auth__box-container
    background-color: white
    border-radius: 2px
    box-shadow: 0px 3px $dark-gray
    padding: 20px
    width: 250px