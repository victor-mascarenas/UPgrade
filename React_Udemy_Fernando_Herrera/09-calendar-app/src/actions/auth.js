import Swal from "sweetalert2";
import { fetchWithoutToken } from "../helpers/fetch";
import { types } from "../types/types";

const storeToken = (token) => {
    localStorage.setItem('token', token);
    localStorage.setItem('token-start', new Date().getTime());
};

export const startLogin = (email, password) => {
    return async (dispatch) => {
        const res = await fetchWithoutToken('auth', {
            email,
            password
        }, 'POST');
        const body = await res.json();
        if (body.ok) {
            storeToken(body.token);
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        } else {
            Swal.fire('Error', body.msg, 'error');
        }
    };
};
export const startRegister = (email, password, name) => {
    return async (dispatch) => {
        const res = await fetchWithoutToken('auth/new', {
            name,
            email,
            password
        }, 'POST');
        const body = await res.json();
        if (body.ok) {
            storeToken(body.token);
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        } else {
            Swal.fire('Error', body.msg, 'error');
        }
    };
};

const login = (user) => ({
    type: types.AUTH_LOGIN,
    payload: user
});