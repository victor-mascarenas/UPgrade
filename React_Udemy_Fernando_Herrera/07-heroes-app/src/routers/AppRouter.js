import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { LoginPage } from '../components/login/LoginPage';

import { DashboardRouter } from './DashboardRouter';

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route path='/login' element={<LoginPage/>}/>
                <Route path='/*' element={<DashboardRouter/>}/>
            </Routes>
        </BrowserRouter>
    );
};
