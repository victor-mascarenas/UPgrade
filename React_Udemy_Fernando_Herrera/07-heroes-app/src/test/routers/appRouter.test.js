import { AppRouter } from "../../routers/AppRouter";
import { mount } from 'enzyme';
import { AuthContext } from "../../auth/authContext";

describe('Pruebas en AppRouter', () => {
    const context = {
        user: {
            logged: false
        }
    };

    test('Debe mostrarse correctamente', () => {
        const wrapper = mount(
            <AuthContext.Provider value={context}>
                <AppRouter/>
            </AuthContext.Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar Login si no esta autenticado', () => {
        const wrapper = mount(
            <AuthContext.Provider value={context}>
                <AppRouter/>
            </AuthContext.Provider>
        );
        expect(wrapper.find('h1').text().trim()).toBe('Login Page');
    });
});