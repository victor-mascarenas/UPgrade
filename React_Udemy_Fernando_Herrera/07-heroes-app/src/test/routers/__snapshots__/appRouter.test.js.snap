// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter Debe mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Page
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
